src/server/api/lib/twilio.js

```diff
@@ -7,45 +7,53 @@ import {
   r,
   cacheableData
 } from "../../models";
-import { log, getConfig } from "../../../lib";
+import { log } from "../../../lib";
 import { saveNewIncomingMessage } from "./message-sending";
+import { symmetricDecrypt } from "./crypto";
 
 // TWILIO error_codes:
 // > 1 (i.e. positive) error_codes are reserved for Twilio error codes
 // -1 - -MAX_SEND_ATTEMPTS (5): failed send messages
 // -100-....: custom local errors
 // -101: incoming message with a MediaUrl
 
-let twilio = null;
 const MAX_SEND_ATTEMPTS = 5;
 const MESSAGE_VALIDITY_PADDING_SECONDS = 30;
 const MAX_TWILIO_MESSAGE_VALIDITY = 14400;
 const DISABLE_DB_LOG = process.env.DISABLE_DB_LOG || global.DISABLE_DB_LOG;
+const SKIP_TWILIO_VALIDATION = process.env.SKIP_TWILIO_VALIDATION || global.SKIP_TWILIO_VALIDATION;
+
+const headerValidator = () => {
+  if (!!SKIP_TWILIO_VALIDATION) return (req, res, next) => next();
+
+  return async (req, res, next) => {
+    const organization = req.params.orgId
+      ? await cacheableData.organization.load(req.params.orgId) : null;
+    const { authToken } = organization
+      ? await getTwilioAuth(organization)
+      : process.env.TWILIO_AUTH_TOKEN;
+    const options = {
+      validate: true,
+      protocol: "https",
+    };
 
-if (process.env.TWILIO_API_KEY && process.env.TWILIO_AUTH_TOKEN) {
-  // eslint-disable-next-line new-cap
-  twilio = Twilio(process.env.TWILIO_API_KEY, process.env.TWILIO_AUTH_TOKEN);
-} else {
-  log.warn("NO GLOBAL TWILIO CONNECTION");
-}
-
-if (!process.env.TWILIO_MESSAGE_SERVICE_SID) {
-  log.warn(
-    "Twilio will not be able to send without TWILIO_MESSAGE_SERVICE_SID set"
-  );
-}
+    return Twilio.webhook(authToken, options)(req, res, next);
+  };
+};
 
-function webhook() {
-  log.warn("twilio webhook call"); // sky: doesn't run this
-  if (twilio) {
-    return Twilio.webhook();
-  } else {
-    log.warn("NO TWILIO WEB VALIDATION");
-    return function(req, res, next) {
-      next();
-    };
+const getTwilioAuth = async organization => {
```

Comment:
So, I think all this logic should be in the organization.getTwilioAuth -- no reason to split up the logic into two separate files.